src/System.Runtime.Extensions/src/System/Environment.Win32.cs

```diff
@@ -341,13 +342,17 @@ public static string UserName
                 }
 
                 return string.Empty;
+#else
+                return "Windows User";
+#endif
```

Comment:
I think the GetUserNameCore() approach is less ugly than this pass by ref approach but tha'ts just me..